webviews/wkwebview/wkwebview/LinkViewController.swift

```diff
@@ -2,138 +2,185 @@
 //  LinkViewController.swift
 //  wkwebview
 //
-//  Copyright (c) 2016 Plaid Inc. All rights reserved.
+//  Copyright (c) 2020 Plaid Inc. All rights reserved.
 //
 
 import UIKit
 import WebKit
 
-class LinkViewController: UIViewController, WKNavigationDelegate {
+final class LinkViewController: UIViewController, WKNavigationDelegate {
+
+    /// `handleRedirctURL` checks the `url` to see if it matches the redirect url used when creating a link token.
+    /// If so, it re-initializes the webview to finish the link session. Returns `true` if handled, `false` otherwise.
+    func handleRedirectURL(_ url: URL) -> Bool {
+        /// This should be the same `redirect_uri` used when creating a link token via https://plaid.com/docs/api/tokens/#linktokencreate
+        let redirectURI = URL(string: "<#YOUR_APPLICATION_REDIRECT_URI#>")!
+
+        // If the loaded url matches the redirectURI, then re-initialize the webview to complete the
+        // authentication flow with the additional `receivedRedirectUri` parameter.
+        if redirectURI.host == url.host && redirectURI.path == url.path && redirectURI.scheme == url.scheme {
```

Comment:
Including a check on `.path` may cause issues with "top-level" `redirectURI`s (e.g. `https://example.net/`) where the trailing slash (`/`) may not be present on the `url.path` passed in from the activity.

Reply:
+1

Reply:
⚡